### ISSUE TITLE
autodoc causes unnecessary rebuilds when a module member is a built-in type

### ISSUE THREAD

**Issue by dzhu** [sphinx-doc/sphinx#5670]
Subject: autodoc causes unnecessary rebuilds when a module member is a built-in type

### Problem
When autodoc is in use, if a module has a member that is a built-in type (not an instance, but the actual type itself), then the module is always believed to have a changed dependency. It looks like the ultimate problem is this: when `sphinx.ext.autodoc.Documenter.generate` is called for that object, `sphinx.pycode.ModuleAnalyzer.for_module` is called for the `builtins` module; that then calls `sphinx.pycode.ModuleAnalyzer.for_string`, which returns an object with `srcname == '<string>'`. Then `'<string>'` goes straight into the set of dependencies in the build environment as a file name; since that file doesn't exist, it always gets marked as changed by `sphinx.environment.BuildEnvironment.get_outdated_files`.

#### Procedure to reproduce the problem
Create a module with a member that is a built-in type (see [example](https://github.com/sphinx-doc/sphinx/files/2612786/example.zip)) and use autodoc to document it, including the `:members:` directive.

#### Error logs / results
Even when `sphinx-build` is run twice in a row with no files changed, the environment is updated, as seen below, causing the build to take longer than it should.
```
Running Sphinx v1.8.2
loading pickled environment... done
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 0 source files that are out of date
updating environment: [] 0 added, 1 changed, 0 removed
<rest snipped>
```

#### Expected results
The cached environment should not be updated if no files have been changed.

### Reproducible project / your project
[example.zip](https://github.com/sphinx-doc/sphinx/files/2612786/example.zip)
Unzip and run `run.sh` twice.

### Environment info
- OS: Linux (Ubuntu 18.04)
- Python version: 3.6.6
- Sphinx version: 1.8.2/1.7.5 both tested

**Comment by tk0miya** [comment]
Oh, I notice this is same as #5614. It was already fixed at 1.8.3.
Please try newer release.
Thanks,

**Comment by dzhu** [comment]
It is indeed fixed. Thank you!
